catchup.sh

```diff
@@ -5,167 +5,171 @@ RED='\033[0;51;30m'
-   conan upload "*" -r=artifactory --all --confirm
+    echo "performing Exercise 9 (Upload packages to artifactory)"
+    conan upload hello/0.1@user/testing -r artifactory --all
+    conan search -r=artifactory
+    conan search hello/0.1@user/testing -r=artifactory
+    conan upload "*" -r=artifactory --all --confirm
 }
 
 explore_cache() {
-   echo "performing Exercise 10 (Explorer cache and remove packages)"
-   ls ~/.conan
-   ls ~/.conan/data
-   ls ~/.conan/data/hello/0.1/user/testing
-   conan remove "*" -f
-   conan search
-   ls ~/.conan/data
+    echo "performing Exercise 10 (Explorer cache and remove packages)"
+    ls ~/.conan
+    ls ~/.conan/data
+    ls ~/.conan/data/hello/0.1/user/testing
+    conan remove "*" -f
+    conan search
+    ls ~/.conan/data
 }
 
 consume_artifactory() {
-   echo "performing Exercise 11 (Consume packages from artifactory)"
-   # remove everything from local cache
-   conan remove "*" -f
-   cd consumer/build
-   conan install .. -r=artifactory
-   cmake .. -DCMAKE_BUILD_TYPE=Release
-   cmake --build .
-   cd bin
-   ./timer
+    echo "performing Exercise 11 (Consume packages from artifactory)"
+    # remove everything from local cache
+    conan remove "*" -f
+    cd consumer/build
+    conan install .. -r=artifactory
+    cmake .. -DCMAKE_BUILD_TYPE=Release
+    cmake --build .
+    cd bin
+    ./timer
 }
 
 create_options_shared() {
-   echo "performing Exercise 12 (Package options: shared)"
-   cd create_sources
-   conan create . user/testing -o hello:shared=True
-   conan create . user/testing -o hello:shared=True -s build_type=Debug
-   conan search hello/0.1@user/testing
+    echo "performing Exercise 12 (Package options: shared)"
+    cd create_sources
+    conan create . user/testing -o hello:shared=True
+    conan create . user/testing -o hello:shared=True -s build_type=Debug
+    conan search hello/0.1@user/testing
 }
 
 create_options_greet() {
-   echo "performing Exercise 13 (Custom options: language)"
-   cd create_options
-   sed -i 's/self.copy2/self.copy/g' conanfile.py
-   conan create . user/testing -o greet:language=English
-   conan create . user/testing -o greet:language=Spanish
+    echo "performing Exercise 13 (Custom options: language)"
+    cd create_options
+    sed -i 's/self.copy2/self.copy/g' conanfile.py
+    conan create . user/testing -o greet:language=English
+    conan create . user/testing -o greet:language=Spanish
 }
 
 configuration_values() {
-   echo "performing Exercise 14 (Configuration values and errors)"
-   cd create_options
-   conan create . user/testing -o greet:language=Italian
-   conan inspect greet/0.1@user/testing
-   conan inspect zlib/1.2.11@
-   conan get zlib/1.2.11@
-
-   conan create . user/testing -s compiler=unknown
-   conan create . user/testing -s compiler.version=200
-   cat ~/.conan/settings.yml
+    echo "performing Exercise 14 (Configuration values and errors)"
+    cd create_options
+    set +e
```

Comment:
Why do we need to change this? Can't be changed from the CI scripts only?

Reply:
This is done to avoid raising when there is an expected error in a specific command. I don't know how to whitelist this without whitelisting the whole exercise execution...